core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala

```diff
@@ -31,20 +34,28 @@ import org.broadinstitute.dsde.rawls.util.ScalaConfig._
 import org.broadinstitute.dsde.rawls.util._
 import org.broadinstitute.dsde.rawls.webservice._
 import org.broadinstitute.dsde.rawls.workspace.{WorkspaceService, WorkspaceServiceConfig}
-import org.broadinstitute.dsde.rawls.config._
 import org.broadinstitute.dsde.workbench.google.GoogleCredentialModes.Json
 import org.broadinstitute.dsde.workbench.google.HttpGoogleBigQueryDAO
-
-import scala.collection.JavaConversions._
+import org.broadinstitute.dsde.workbench.google2._
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import org.broadinstitute.dsde.workbench.util.ExecutionContexts
+import org.http4s.Uri
+import org.http4s.client.blaze.BlazeClientBuilder
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
-import scala.util.{Failure, Success}
-import net.ceedubs.ficus.Ficus._
-import org.apache.commons.io.FileUtils
-import org.broadinstitute.dsde.workbench.model.google.GcsBucketName
 
-object Boot extends App with LazyLogging {
-  private def startup(): Unit = {
+object Boot extends IOApp with LazyLogging {
+  override def run(
+                    args: List[String]
+                  ): IO[ExitCode] =
```

Comment:
I am concerned with adding IO to rawls which operates with Future. Can what you're doing in the PR not be done without using IOs?

Reply:
I don't think so. One thing is I'm uisng `cats.effect.Resource` for resource management(making sure resources are released when they're done), and `Future` doesn't work well with it. `google2` also only exposes `F` interfaces, which can't be `Future`